dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -633,9 +634,30 @@ public String getKey()
             .add( "timeField", timeField )
             .add( "orgUnitField", orgUnitField )
             .add( "userOrgUnitType", userOrgUnitType )
+            .add( "dimensionItemKeywords", getDimensionalItemKeywords( filters ) )
             .addIgnoreNull( "apiVersion", apiVersion ).build();
     }
 
+    private String getDimensionalItemKeywords( final List<DimensionalObject> filters )
```

Comment:
Just checking, would it be better to include this in `BaseDimensionalObject`#`getKey` method by including `DimensionItemKeywords` if not null? (Instead of iterating over this again).

Reply:
Unfortunately at that point, we don't know `DimensionItemKeywords` in BaseDimensionalObject.
To avoid an extra iteration, I added it as part of the original iteration of the filter and dimensions (first lines on the method `getKey()` in DataqueryParams.